src/speed_challenge_ctrl.cpp

```diff
@@ -28,18 +52,207 @@ class SpeedChallenge{
         }
     }
 
+    float error = 2;
+    bool isReached() {
+
+        bool atDestination = false;
+
+        // TODO: use conversion node 
+        // Converting to gazebo coordinates 
+        float current_pos_x = current_pos_.pose.position.y;
+        float current_pos_y = -1 * current_pos_.pose.position.x;
+
+        ROS_INFO_STREAM(current_pos_);
+
+        if (current_pos_x < task_goal_pos.point.x + error & current_pos_x > task_goal_pos.point.x - error) {
+            if (current_pos_y < task_goal_pos.point.y + error & current_pos_y > task_goal_pos.point.y - error) {
+                atDestination = true;
+            }
+        }
+
+        return atDestination;
+    }
+
 private:
 
     ros::NodeHandle nh_;
     ros::NodeHandle private_nh_;
+
     ros::Subscriber prop_sub;
+    ros::Subscriber task_to_exec;
+    ros::Subscriber global_pos_;
+    ros::Publisher task_pos;
+   
+    //Params 
+    float avoidance_angle;
+    float avoidance_distance;
+    float acceptable_error;
+
+    std::string local_pose_topic_;
 
-    prop_mapper::PropArray prop_array_test;
-    prop_mapper::PropArray prop_array;
+    geometry_msgs::PoseStamped current_pos_;
 
+    // Direction vectors
+    std::vector<float> starting_pos_2_prop;
+    std::vector<float> point1_direction_vec;
+    std::vector<float> point2_direction_vec;
```

Comment:
Why are vectors being used here? Is it to store x and y together? If so, it might be clearer to use geometry_msgs::Point and ignore the z, or have a separate variable for x and y.